File: conversions/csharp_unity/Trait.cs
using System;
using System.Collections.Generic;
[System.Serializable]
public class Trait : BaseElement
{
    // Qualitative
    public string SocialEffects;
    public string PhysicalEffects;
    public string SkillEffects;
    public string PersonalityEffects;
    public string ArtisticEffects;
    public string BehaviourEffects;
    // Quantitative
    public int? Charisma;
    public int? Coercion;
    public int? Capability;
    public int? Compassion;
    public int? Creativity;
    public int? Courage;
    // World
    public string AntiTraitId;
    public List<string> EmpoweredAbilitiesIds; = new List<string>();
}

File: languages/typescript/Trait.ts
import { BaseElement } from './base_elements';

export interface Trait extends BaseElement {
  // Qualitative
  socialEffects?: string | null;
  physicalEffects?: string | null;
  skillEffects?: string | null;
  personalityEffects?: string | null;
  artisticEffects?: string | null;
  behaviourEffects?: string | null;
  // Quantitative
  charisma?: number | null;
  coercion?: number | null;
  capability?: number | null;
  compassion?: number | null;
  creativity?: number | null;
  courage?: number | null;
  // World
  antiTraitId?: string | null;
  empoweredAbilitiesIds?: string[] | null;
}
